### PR TITLE
Add help page

### PR DESCRIPTION
**Pull request by rkeulemans**
Solves #314

### CODE REVIEW

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            return node.x;
+        })
+        .attr("y", (1/2) * labelHeight + nodeRadius + labelOffset)
+        .text(function (node) {
+            return node.name;
+        });
+}
+
+function drawNodeSizeSvg () {
+    var svgNodeSize = d3.select("#help-circle-size");
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: You can use `var` only once and then the different variables separated with comma's.

`var x = 5,
       y = 6;`

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+            return edge.x;
+        })
+        .attr("y", (1/2) * labelHeight + maxEdgeWidth + labelOffset)
+        .text(function (edge) {
+            return edge.name;
+        });
+}
+
+function drawDividingLineSvg () {
+    var svgConnectionSizeContainer = d3.select("#help-connection-balance");
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: See the other comment about `var`

File: TriblerGUI/widgets/trustpage/js/style_config.js [hunk 3]
```diff
@@ -59,7 +60,29 @@ var config = {
...
+        numberOfEdges: 3,
+        labelOffset: 5,
+        circleColor : {
+            nodeRadius: 20,
+            colorNames: ["Freerider", "Unreliable", "Neutral", "Reliable", "Trusted"]
+        },
+        circleSize : {
+            minNodeRadius: 15,
+            nodeRadiusStep: 2,
+            sizeNames: ["least", "less", "average", "more", "most"]
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/style_config.js]: It might be worth capitalising the size, width and lineNames

**Comment by vandenheuvel** [comment]
retest this please

**Comment by MaxVanDeursen** [comment]
PS. The JavaScript tests are failing because you use d3 functions in utilities.py. This library is not imported in the tests.

File: TriblerGUI/widgets/trustpage/index.html [hunk 4]
```diff
@@ -10,17 +10,35 @@
...
 <svg id="graph" width="600" height="600" viewBox="0 0 600 600"></svg>
-<div class="tooltip">
-    <div id="tooltip-content">
-        <p class="header">Nodes</p>
+<div id="help-overlay" onclick="closeHelpPage()"></div>
+<div id="help-button" onclick="showHelpPage()">Help</div>
+<div id="help-page">
+    <div id="help-page-content">
+        <h2 id="help-page-title">Visualization of the Tribler network</h2>
+        <p>In the Tribler network, users exchange content anonymously by uploading and downloading. This only works if everyone makes content available by uploading. Users that decide not to upload at least as much as they download, are blocked from downloading until they start uploading again.</p>
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/index.html]: I'd write: "Users who download more than they upload are automatically banned by all other users until they upload more."

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
+function showHelpPage () {
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Please put this file in the correct format:
```
function RadialHelp() {
    var self = this;

   self.methodName = function () { // method body }
}
```

Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Furthermore, make a method .initialize() in which you draw the entire help page (but don't display it). This way it won't be redrawn every time you click 'Help'

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+    document.getElementById("help-button").innerHTML = "Close";
+    document.getElementById("help-overlay").style.display = "block";
+    document.getElementById("help-page").style.display = "block";
+    document.getElementById("graph").style.opacity = "0.5";
+    drawSvgs();
+}
+
+function closeHelpPage () {
+    document.getElementById("help-button").innerHTML = "Help";
+    document.getElementById("help-overlay").style.display = 'none';
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: All these .style things should be in css.

```
.help-page{
   display:none;
   ..
}
.help-page.visible {
   display: block;
   ...
}
Then using javascript you add a class .visible to the help-page. Then css will make the window visible.

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+function drawSvgs () {
+    drawIdentitySvg();
+    drawReputationSvg();
+    drawNodeSizeSvg();
+    drawConnectionSizeSvg();
+    drawDividingLineSvg();
+}
+
+function drawIdentitySvg () {
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Nice separation of drawing logic, please add as private methods in the new format _drawIdentitfySvg()

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+function drawReputationSvg () {
+    var svgNodeColor = d3.select("#help-circle-color"),
+
+        groups = svgNodeColor.append("svg")
+        .attr("id", "reputation-legend")
+        .attr("x", 0)
+        .attr("y", parseInt(svgNodeColor.style("height"), 10) / 2)
+        .attr("overflow", "visible"),
+
+        nodeRadius = config.help.circleColor.nodeRadius,
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Basic javascript: when declaring a variable you use 'var'

File: TriblerGUI/widgets/trustpage/js/utilities.js [hunk 9]
```diff
@@ -87,6 +92,13 @@ function ucfirst(str) {
 }
 
 /**
+ * Get node color based on node score [0, 1]
+ */
+var nodeColor = d3.scaleLinear()
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/utilities.js]: Move to help: self._nodeColor = ...

File: TriblerGUI/widgets/trustpage/index.html [hunk 10]
```diff
@@ -6,22 +6,41 @@
...
         </ul>
-        <p class="header">Edges</p>
+        <h3>Exchanged data as lines</h3>
+        <p>Exchanged data between users is depicted by lines. Each line has:</p>
         <ul>
-            <li>Size: the relative size of the upload quantity to all available.</li>
-            <li>Division point: the ratio between the amounts uploaded between two nodes.</li>
+            <li>A width representing the amount of content exchanged between these users</li>
+            <svg id="help-connection-transmission" class="help-svg"></svg>
+            <li>A dividing indicator indicating which user uploaded more</li>
```
Comment (vandenheuvel) [inline TriblerGUI/widgets/trustpage/index.html]: How about "A marker indicating which user uploaded more data".

Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/index.html]: Not really a "marker" in my opinion.

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+        self.visible = true;
+    };
+
+    self.close = function () {
+        document.getElementById("help-button").innerHTML = "Help";
+        document.getElementsByTagName("body")[0].className = "";
+        self.visible = false;
+    };
+
+    self.visible = false;
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Move this 'state' attribute to the top, it feels lonely between these methods

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+    self.toggle = function () {
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Please add a docstring

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+        self._drawLineDividerSvg();
+    };
+
+    self.show = function () {
+        document.getElementById("help-button").innerHTML = "Close";
+        document.getElementsByTagName("body")[0].className = "show-help-page";
+        self.visible = true;
+    };
+
+    self.close = function () {
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: Please add a doctstring

File: TriblerGUI/widgets/trustpage/js/help_page.js [hunk 14]
```diff
@@ -0,0 +1,260 @@
...
+            numberOfNodes = self.config.help.numberOfNodes,
+            labelOffset = self.config.help.label.offsetY,
+            width = parseInt(svg.style("width")),
+            spacing = width / (numberOfNodes + 1),
+            labelY = (1 / 2) * nodeRadius + labelOffset + self.config.help.label.fontSize,
+            nodesData = [];
+
+        // Fill array with the data belonging to each node
+        for (var i = 0; i < numberOfNodes; i++) {
+            nodesData.push({
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/help_page.js]: As we have done in the methods below, simply put the append in this for loop. You can then remove the whole nodesData array. Take a good look at how we have done this below.

Together with the ```drawCircleSizeSvg``` you could move this to a separate ```_drawNode``` method

File: TriblerGUI/widgets/trustpage/js/style_config.js [hunk 15]
```diff
@@ -70,7 +71,36 @@ var config = {
...
+        numberOfNodes: 5,
+        numberOfEdges: 3,
+        circleColor : {
+            nodeRadius: 20,
+            colorNames: ["Freerider", "Unreliable", "Neutral", "Reliable", "Trusted"]
+        },
+        circleSize : {
+            sizeNames: ["Least", "Less", "Average", "More", "Most"]
+        },
+        edgeWidth: {
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/style_config.js]: This is not just about the edgeWidth svg. It also contains some config fields for the edgeSeparator svg.

File: TriblerGUI/widgets/trustpage/js/style_config.js [hunk 16]
```diff
@@ -70,7 +71,36 @@ var config = {
...
+            fontWeight: "bold",
+            color: "#FFFFFF",
+            offsetY: 10
+        },
+        examplePublicKey: "#bcb",
+        numberOfNodes: 5,
+        numberOfEdges: 3,
+        circleColor : {
+            nodeRadius: 20,
+            colorNames: ["Freerider", "Unreliable", "Neutral", "Reliable", "Trusted"]
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/style_config.js]: why not just all it 'labels' ?

File: TriblerGUI/widgets/trustpage/js/style_config.js [hunk 17]
```diff
@@ -70,7 +71,36 @@ var config = {
...
+        },
+        circleSize : {
+            sizeNames: ["Least", "Less", "Average", "More", "Most"]
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/style_config.js]: rename to labels

File: TriblerGUI/widgets/trustpage/js/style_config.js [hunk 18]
```diff
@@ -70,7 +71,36 @@ var config = {
...
+        },
+        edgeWidth: {
+            edgeLength: 100,
+            dividingWidth: 2,
+            widthNames: ["Little", "Average", "Much"],
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/style_config.js]: rename to labels

File: TriblerGUI/widgets/trustpage/js/style_config.js [hunk 19]
```diff
@@ -70,7 +71,36 @@ var config = {
...
+            lineNames: ["Right user", "Balanced", "Left user"]
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/style_config.js]: rename to labels (after moving to separate config struct)

**Comment by vandenheuvel** [comment]
retest this please

File: TriblerGUI/widgets/trustpage/help_page.css [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+
+#help-page-content {
+    padding: 10px;
+}
+
+#help-page-title {
+    text-align: center;
+}
+
+::-webkit-scrollbar-track
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/help_page.css]: Cross browser compatibility will be a separate PR that covers all code.
